Client/Frontend/Browser/SearchEngines.swift

```diff
@@ -140,6 +163,21 @@ class SearchEngines {
         }
     }
 
+    class func customEngineFilePath() -> String {
+        let fileManager = NSFileManager.defaultManager()
+        let url = fileManager.URLsForDirectory(.DocumentDirectory, inDomains: .UserDomainMask)[0]
```

Comment:
We should probably use the profile `fileAccessor` here (by adding it as another `init` parameter)  to use a document directory that's relative to the profile instead of the device document directory. That way, we won't clobber your custom search engines when running tests, for example!